presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushPartialAggregationThroughExchange.java

```diff
@@ -71,11 +71,22 @@ public PushPartialAggregationThroughExchange(FunctionRegistry functionRegistry)
     @Override
     public Result apply(AggregationNode aggregationNode, Captures captures, Context context)
     {
+        PlanNode childNode = context.getLookup().resolve(aggregationNode.getSource());
+        if (!(childNode instanceof ExchangeNode)) {
+            return Result.empty();
+        }
+
+        ExchangeNode exchangeNode = (ExchangeNode) childNode;
+
         boolean decomposable = aggregationNode.isDecomposable(functionRegistry);
 
         if (aggregationNode.getStep().equals(SINGLE) &&
```

Comment:
In this form this condition seems redundant to https://github.com/prestodb/presto/pull/10224/files#diff-b7cdbeaf2c1b99719da7ae1d1bb76ca7R131, no? However, it's more explicit with a nice comment so we might as well have it. 

However, even it this condition evaluates to false via `exchangeNode.getType() != REPARTITION` I think it will still be split (correctly) for gathering exchange via https://github.com/prestodb/presto/pull/10224/files#diff-b7cdbeaf2c1b99719da7ae1d1bb76ca7R13

Reply:
(somehow those links don't work for me. They just load the full PR. I assume you're trying to show me a specific line?)

Reply:
This code:
```
         switch (aggregationNode.getStep()) {
             case SINGLE:
                 // Split it into a FINAL on top of a PARTIAL and
return Result.ofPlanNode(split(aggregationNode, context));
```
from this file: `PushPartialAggregationThroughExchange.java`